scripts/file_collector.py

```diff
@@ -0,0 +1,187 @@
+"""
+The purpose of this script is to collect all the /storage/ filepaths used by
+the p2a_impacts package.
+"""
+
+from argparse import ArgumentParser
+from sqlalchemy import create_engine
+from sqlalchemy.orm import sessionmaker
+import logging
+
+from ce.api.util import search_for_unique_ids
+from modelmeta import DataFile
+from p2a_impacts.parser import build_parse_tree
+from p2a_impacts.evaluator import evaluate_rule
+from p2a_impacts.fetch_data import (
+    get_dict_val,
+    read_csv,
+    translate_args,
+    get_models,
+)
+from p2a_impacts.utils import get_region, REGIONS
+
+
+def setup_logging(log_level):
+    formatter = logging.Formatter(
+        "%(asctime)s %(levelname)s: %(message)s", "%Y-%m-%d %H:%M:%S"
+    )
+    handler = logging.StreamHandler()
+    handler.setFormatter(formatter)
+    logger = logging.getLogger("scripts")
+    logger.addHandler(handler)
+    logger.setLevel(getattr(logging, log_level))
+    return logger
+
+
+logger = setup_logging("INFO")
+
+
+def file_collection(csv, date_range, region, ensemble, connection_string):
+    """
+    Builds the variables that would be used in in p2a_impacts.resolve_rules
+    to the point of accessing the climate explorer database, but instead of evaluating
+    a rule it writes the paths for the files used to a text file.
+    """
+
+    # read csv
+    logger.info("Reading {}".format(csv))
+    rules = read_csv(csv)
+
+    # create parse tree dictionary and gather unique variables
+    logger.info("Building parse tree")
+    parse_trees = {}
+    variables = {}
+    region_variable = None
+    for rule, condition in rules.items():
+        try:
+            parse_trees[rule], vars, region_var = build_parse_tree(condition)
+        except SyntaxError as e:
+            logger.warning("{}, rule will be excluded".format(e))
+            continue
+
+        # check region var
+        if region_var:
+            region_variable = region_var
+
+        # add unique variables to set
+        for name, values in vars.items():
+            if name not in variables.keys():
+                variables[name] = values
+
+    # get values for all variables we would need if we were doing an evaluation
+    logger.info("Collecting variables")
+    Session = sessionmaker(create_engine(connection_string))
+    sesh = Session()
```

Comment:
Much of this code also already exists. I know this is what we discussed but seeing it now makes me wonder if there would be a way to refactor this. For now it is probably ok, but just something to keep in mind.